core/services/fluxmonitor/flux_monitor.go

```diff
@@ -438,27 +438,25 @@ func (p *PollingDeviationChecker) Start() {
 
 func (p *PollingDeviationChecker) setIsHibernatingStatus() {
 	if p.flagsContract != nil {
-		isHibernating, err := p.isFlagRaised()
-		if err != nil {
-			panic(err)
-		}
-		p.isHibernating = isHibernating
+		isFlagLowered, err := p.isFlagLowered()
+		logger.ErrorIf(err, "unable to set hibernation status")
+		p.isHibernating = !isFlagLowered
```

Comment:
Hmmm do you really want to change the flag status if this fails?

Reply:
we basically default to "normal mode" if `isFlagLowered()` returns an error. But that doesn't seem obvious by looking at this code... refactoring.